### ISSUE TITLE
Serve nested metadata

### ISSUE THREAD

**Issue by urzds** [coreos/coreos-baremetal#84]
When defining nested metadata like the following:
```yaml
---
api_version: v1alpha1
groups:
  - name: Worker
    spec: worker
    metadata:
      fleet:
        public_ip: ...
        metadata: ...
```

bootcfg will generate the following on requests to `/metadata`:
```sh
FLEET=map[public_ip:...]
```

I would instead expect the following:
```sh
FLEET_PUBLIC_IP=...
FLEET_METADATA=...
```

**Comment by dghubble** [comment]
Seems like a reasonable expectation. I've seen double `__` used as a separator in other contexts, if reversibility is a concern. Also, there is the question of how best to handle metadata containing lists.

**Comment by urzds** [comment]
lists: I'd be lazy and just skip them for now. Or create a space separated string variable.
double underscores: That would make it impossible to generate CoreOS systemd.service compatible names, like FLEET_METADATA.

I am basically trying to save me some typing and to make the bootcfg file easier to grasp by eliminating common prefixes (which incidentally correspond with the variable prefixes expected by CoreOS systemd.service names).

**Comment by dghubble** [comment]
Closed by #279
